### PR TITLE
chore: support for disable automated sync

### LINKED ISSUE #255: Support disabling automated sync at cluster level
# Summary

What change needs making?

To support stop automated sync at the managed cluster level.

When would you use this?

Code freeze.

---

**Message from the maintainers**:

If you wish to see this enhancement implemented please add a 👍 reaction to this issue! We often sort issues this way to know what to prioritize.

### PR DESCRIPTION
**Pull request by xdevxy**
Fixes #255 

### Modifications

Support disabling automated sync for all gitsyncs in the cluster.

### CODE REVIEW

File: internal/controller/config/config.go [hunk 1]
```diff
@@ -34,11 +34,12 @@ func GetConfigManagerInstance() *ConfigManager {
...
-	ClusterName        string `json:"clusterName" mapstructure:"clusterName"`
-	SyncTimeIntervalMs int    `json:"syncTimeIntervalMs" mapstructure:"syncTimeIntervalMs"`
-	CascadeDeletion    bool   `json:"cascadeDeletion" mapstructure:"cascadeDeletion"`
-	AutoHealEnabled    bool   `json:"autoHealEnabled" mapstructure:"autoHealEnabled"`
-	IncludedResources  string `json:"includedResources" mapstructure:"includedResources"`
+	ClusterName           string `json:"clusterName" mapstructure:"clusterName"`
+	SyncTimeIntervalMs    int    `json:"syncTimeIntervalMs" mapstructure:"syncTimeIntervalMs"`
+	AutomatedSyncDisabled bool   `json:"automatedSyncDisabled" mapstructure:"automatedSyncDisabled"`
+	CascadeDeletion       bool   `json:"cascadeDeletion" mapstructure:"cascadeDeletion"`
+	AutoHealDisabled      bool   `json:"autoHealDisabled" mapstructure:"autoHealDisabled"`
```
Comment (xdevxy) [inline internal/controller/config/config.go:41]: Updated auto heal flag to match the same format as automated sync.

File: internal/sync/syncer.go [hunk 2]
```diff
@@ -246,7 +248,7 @@ func (s *Syncer) runOnce(ctx context.Context, key string, worker int) error {
 	}
 	// If auto heal is not enabled and the target commit hash is the same as last sync,
 	// skip the syncing.
-	if !autoHeal && lastSyncedCommitHash != nil && lastSyncedCommitHash.Hash == commitHash {
+	if !globalConfig.AutoHealDisabled && lastSyncedCommitHash != nil && lastSyncedCommitHash.Hash == commitHash {
```
[the remark this answers is not in the document]
Comment (xdevxy) [reply internal/sync/syncer.go]: Ah good catch, thanks!

**Comment by juliev0** [comment]
Actually, should we be doing this on an individual file basis?

If auto heal is off, I'm a user, and I manually update my `Deployment` resource in the cluster but then I change my `ConfigMap` spec in github, I want my `ConfigMap` to get updated but I don't want my `Deployment` to be auto-healed.

**Comment by xdevxy** [comment]
> Actually, should we be doing this on an individual file basis?
> 
> If auto heal is off, I'm a user, and I manually update my `Deployment` resource in the cluster but then I change my `ConfigMap` spec in github, I want my `ConfigMap` to get updated but I don't want my `Deployment` to be auto-healed.

Yeah, can you create an issue to track this feature? Thanks!

**Comment by juliev0** [comment]
> Yeah, can you create an issue to track this feature? Thanks!

Done. I guess I was more thinking it was a Bug - not sure if you agree?
https://github.com/numaproj-labs/numaplane/issues/261

**Comment by xdevxy** [comment]
> > Yeah, can you create an issue to track this feature? Thanks!
> 
> Done. I guess I was more thinking it was a Bug - not sure if you agree? #261

Not quite agree, to me this is a feature to auto heal at individual resource level instead of whole gitsync.

**Review by juliev0 — APPROVED** [review]

**Pull request merged**